torch/csrc/jit/passes/frozen_ops_to_mkldnn.cpp

```diff
@@ -77,6 +78,63 @@ Operation ConstantMKLDNNTensorOp(const Node* node) {
   };
 }
 
+// aten::convolution does a lot of precomputation and dispatching before
+// mkldnn_convolution is called. registering here we can directly invoke the op
+// and avoid overhead. avoiding dispatch overhead for other operators - relu,
+// add, etc - did not benchmark as speeding up models noticeably. the additional
+// overhead of `convolution` warrants the custom operator.
+jit::RegisterOperators reg_fut_ops({
+    jit::Operator(
+        // XXX: this follows the schema convention of conv2d/conv3d, not
+        // aten::mkldnn_convolution, which is different for some reason!
+        "prim::mkldnn_convolution(Tensor input, Tensor weight, Tensor? bias, int[] stride, int[] padding, int[] dilation, int groups) -> Tensor",
+        [](jit::Stack* stack) {
+          int64_t groups = pop(stack).toInt();
+          auto dilation = pop(stack).toIntVector();
+          auto padding = pop(stack).toIntVector();
+          auto stride = pop(stack).toIntVector();
+
+          Tensor bias;
+          IValue bias_ival = pop(stack);
+          if (!bias_ival.isNone()) {
+            bias = bias_ival.toTensor();
+          }
+          Tensor weight = pop(stack).toTensor();
+          Tensor input = pop(stack).toTensor();
+
+          at::AutoNonVariableTypeMode non_var_type_mode(true);
+          // aten::convolution takes care of 0 dim case before calls into
+          // backends
+          if (input.size(0) == 0) {
```

Comment:
it's weird that we only check `size(0)` but it seems you borrowed it from the convolution implementation.

Reply:
yep, this is coming from convolution.cpp